### ISSUE TITLE
Download CSV from HTML report?

### ISSUE THREAD

**Issue by gwideman** [allinurl/goaccess#1073]
I see that in the HTML report, in the left "gear" menu there's a function to export the data as JSON.

Is there a way to invoke the equivalent functionality, but as CSV?

For what it's worth, I think the customer to whom it's useful to present the nice HTML view is more likely to be able to wrangle CSV (in Excel) than JSON.  At least it is in my case at hand :-).

**Comment by allinurl** [comment]
I agree, having a csv option is useful. Let me take a look at it, it think it shouldn't be to difficult to generate this from the JSON feed, though the only concern is that the JSON object is not completely flat.

**Comment by gwideman** [comment]
Thanks for your response! A couple of suggestions:

1. Part of the impetus for a CSV export is to be able to browse a panel's data all at once, instead of 10 rows at a time. So a convenient UX for that desire could be a pop up view of only that one panel, expanded to show all rows (and including the same sort/filter features of the current panel).

2. There certainly are separate motivations to obtain CSVs, for import into Excel or other tools. If non-flatness of the complete page's data presents an obstacle, might it make sense to have CSV download as a feature of the individual panel rather than the page?
